crypto/merkle/proof_test.go

```diff
@@ -133,6 +133,70 @@ func TestProofOperators(t *testing.T) {
 	assert.NotNil(t, err)
 }
 
+func TestProofOperatorsFromKeys(t *testing.T) {
+	var err error
+
+	// ProofRuntime setup
+	// TODO test this somehow.
+
+	// ProofOperators setup
+	op1 := NewDominoOp("KEY1", "INPUT1", "INPUT2")
+	op2 := NewDominoOp("KEY%2", "INPUT2", "INPUT3")
+	op3 := NewDominoOp("", "INPUT3", "INPUT4")
+	op4 := NewDominoOp("KEY/4", "INPUT4", "OUTPUT4")
+
+	// add characters to the keys that would otherwise result in bad keypath if
+	// processed
+	keys1 := [][]byte{bz("KEY/4"), bz("KEY%2"), bz("KEY1")}
+	badkeys1 := [][]byte{bz("WrongKey"), bz("KEY%2"), bz("KEY1")}
+	keys2 := [][]byte{bz("KEY3"), bz("KEY%2"), bz("KEY1")}
+	keys3 := [][]byte{bz("KEY2"), bz("KEY1")}
+
+	// Good
+	popz := ProofOperators([]ProofOperator{op1, op2, op3, op4})
+	err = popz.VerifyFromKeys(bz("OUTPUT4"), keys1, [][]byte{bz("INPUT1")})
+	assert.Nil(t, err)
+
+	// BAD INPUT
+	err = popz.VerifyFromKeys(bz("OUTPUT4"), keys1, [][]byte{bz("INPUT1_WRONG")})
+	assert.NotNil(t, err)
```

Comment:
[nit][question] was [`assert.Error`](https://pkg.go.dev/github.com/stretchr/testify@v1.8.1/assert#Error) / [`assert.NoError`](https://pkg.go.dev/github.com/stretchr/testify@v1.8.1/assert#NoError) intentionally not used? They seem to functionally do the same thing as `assert.Nil` / `assert.NotNil` so it may just be a developer preference.

Reply:
just copy pasted the other tests :eyes: , but yeah I don't see why we shouldn't use those